src/main/java/org/folio/fqm/service/PermissionsService.java

```diff
@@ -32,14 +32,22 @@ public class PermissionsService {
   private final ModPermissionsClient modPermissionsClient;
   private final Cache<TenantUserPair, Set<String>> cache = Caffeine.newBuilder().expireAfterWrite(cacheDurationSeconds, TimeUnit.SECONDS).build();
   private final EntityTypeFlatteningService entityTypeFlatteningService;
+  private static final List<String> CROSS_TENANT_FQM_PERMISSIONS = List.of(
+    "fqm.entityTypes.item.get",
+    "fqm.query.async.results.get",
+    "fqm.query.async.post"
+  );
 
-  public Set<String> getUserPermissions() {
-    TenantUserPair key = new TenantUserPair(context.getTenantId(), context.getUserId());
-    return cache.get(key, k -> isEureka ? getUserPermissionsFromRolesKeycloak(k.userId()) : getUserPermissionsFromModPermissions(k.userId()));
+  public Set<String> getUserPermissions(String tenantId) {
+    String tenantIdForQuery = tenantId != null ? tenantId : context.getTenantId();
```

Comment:
If we have a way to handle null tenant IDs, why not just always get it from the context instead of using a parameter? If we do need to pass it in, I'd probably overload the method so that we have a version without the tenant ID parameter instead of having to explicitly pass in `null`

Reply:
We have to pass the tenant ID, because it won't always come from the execution context. The execution context here will correspond to the central tenant, but the tenant ID needed here is for the other ECS tenants. I can implement it with an overloaded method though.